crates/bevy_input/src/gamepad.rs

```diff
@@ -1032,25 +860,40 @@ impl AxisSettings {
         self.threshold
     }
 
-    fn filter(&self, new_value: f32, old_value: Option<f32>) -> Option<f32> {
-        let new_value =
-            if self.deadzone_lowerbound <= new_value && new_value <= self.deadzone_upperbound {
-                0.0
-            } else if new_value >= self.livezone_upperbound {
-                1.0
-            } else if new_value <= self.livezone_lowerbound {
-                -1.0
-            } else {
-                new_value
-            };
-
-        if let Some(old_value) = old_value {
-            if (new_value - old_value).abs() <= self.threshold {
-                return None;
-            }
+    /// Clamps the `raw_value` according to the `AxisSettings`.
+    fn clamp(&self, new_value: f32) -> f32 {
+        if self.deadzone_lowerbound <= new_value && new_value <= self.deadzone_upperbound {
+            0.0
+        } else if new_value >= self.livezone_upperbound {
+            1.0
+        } else if new_value <= self.livezone_lowerbound {
+            -1.0
+        } else {
+            new_value
+        }
+    }
+
+    /// Determines whether the change from `old_value` to `new_value` should
+    /// be registered as a change, according to the `AxisSettings`.
+    fn should_register_change(&self, new_value: f32, old_value: Option<f32>) -> bool {
+        if old_value.is_none() {
+            return true;
         }
 
-        Some(new_value)
+        f32::abs(new_value - old_value.unwrap()) > self.threshold
+    }
+
+    /// Filters the `new_value` based on the `old_value`, according to the `AxisSettings`.
+    ///
+    /// Returns the `AxisSettings::clamp`ed `new_value` if the change exceeds the settings threshold,
```

Comment:
Nit: Perhaps to avoid the "`AxisSettings::clamp`ed", we could say something like:
```suggestion
    /// Returns the [clamped](`AxisSettings::clamp`) `new_value` if the change exceeds the settings threshold,
```

Reply:
Agreed!